substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/SerializationSupport.java

```diff
@@ -0,0 +1,319 @@
+            Object reflFactoryInstance = getReflectionFactoryMethod.invoke(null);
+            Method newConstructorForSerializationMethod = ReflectionUtil.lookupMethod(reflectionFactoryClass, "newConstructorForSerialization", Class.class);
+            buildTimeConstructor = (Constructor<?>) newConstructorForSerializationMethod.invoke(reflFactoryInstance, serializationTargetClass);
+
+            // Calculate GeneratedSerializationConstructor for StubForAbstractClass only once
+            if (isAbstract && stubAccessor == null) {
+                stubConstructor = (Constructor<?>) newConstructorForSerializationMethod.invoke(reflFactoryInstance, StubForAbstractClass.class);
+            }
+        } catch (ReflectiveOperationException e) {
+            throw VMError.shouldNotReachHere(e);
+        }
+        if (buildTimeConstructor == null) {
+            return null;
+        }
+        buildTimeConsClass = buildTimeConstructor.getDeclaringClass();
+
+        if (isAbstract && stubAccessor != null) {
+            constructorAccessor = stubAccessor;
+            targetClassName = StubForAbstractClass.class.getName();
+        } else {
+            // Prepare build time checksum and verify with configured checksum only for non-abstract
+            // classes. Abstract class' checksum is always 0.
+            if (!isAbstract) {
+                checksum = getChecksum(serializationTargetClass, configuredChecksum, (FeatureImpl.BeforeAnalysisAccessImpl) access, buildTimeConsClass, targetClassName);
+            }
+            try {
+                Method getConstructorAccessor = ReflectionUtil.lookupMethod(Constructor.class, "getConstructorAccessor");
+                constructorAccessor = getConstructorAccessor.invoke(isAbstract ? stubConstructor : buildTimeConstructor);
+                if (isAbstract) {
+                    assert constructorAccessor != null;
+                    stubAccessor = constructorAccessor;
+                    targetClassName = StubForAbstractClass.class.getName();
+                }
+            } catch (Exception e) {
+                throw VMError.shouldNotReachHere(e);
+            }
+        }
+        // Cache constructorAccessor
+        CachedEntity exitingEntity = cachedSerializationConstructorAccessors.putIfAbsent(targetClassName,
+                        new CachedEntity(constructorAccessor, checksum));
+        if (exitingEntity != null && exitingEntity.configuredChecksum != checksum) {
+            StringBuilder sb = new StringBuilder();
+            sb.append("Suspicious multiple-classloader usage is detected from serialization configurations:\n");
+            sb.append("Serialization target class (name=").append(targetClassName).append(", checksum=").append(checksum).append(")");
+            sb.append(" is already registered with checksum ").append(exitingEntity.configuredChecksum);
+            reportError((FeatureImpl.BeforeAnalysisAccessImpl) access, MULTIPLE_CHECKSUMS, sb.toString());
+        }
```

Comment:
@ziyilin please also remove this bit. We have to allow overlapping/redundant entries in serialization-config.json files.

Reply:
I've loosed the check by adding `exitingEntity.configuredChecksum != checksum`. Only the same name class with different checksum will be considered as error. 
I'm afraid we can't report multi-classloader usage error if the whole check is removed.

Reply:
Ok. That should also do it. Thanks!